packages/metal-quartz-button/src/MetalQuartzButton.soy

```diff
@@ -0,0 +1,47 @@
+{namespace MetalQuartzButton}
+
+/**
+ * This renders the component's whole content.
+ */
+{template .render}
+	{@param? block: bool}
+	{@param? disabled: bool}
+	{@param? href: string}
+	{@param? label: html}
+	{@param? name: string}
+	{@param? size: string}
+	{@param? style: string}
+	{@param? type: string}
+	{@param? value: string}
+	{let $variant: $href ? 'link' : 'button' /}
+	{delcall MetalQuartzButton.markup variant="$variant" data="all" /}
+{/template}
+
+{deltemplate MetalQuartzButton.markup variant="'button'"}
+	{@param? block: bool}
+	{@param? disabled: bool}
+	{@param? label: html}
+	{@param? name: string}
+	{@param? size: string}
+	{@param? style: string}
+	{@param? type: string}
+	{@param? value: string}
+	<button {call MetalQuartzButtonAttributes.attributes data="all" /}>
+		{$label or ''}
+	</button>
+{/deltemplate}
+
+{deltemplate MetalQuartzButton.markup variant="'link'"}
+	{@param? block: bool}
+	{@param? disabled: bool}
+	{@param? href: string}
+	{@param? label: html}
+	{@param? name: string}
+	{@param? size: string}
+	{@param? style: string}
+	{@param? type: string}
+	{@param? value: string}
+	<a href="{$href or ''}" {call MetalQuartzButtonAttributes.attributes data="all" /}>
+		{$label or ''}
+	</a>
+{/deltemplate}
```

Comment:
Looking from the point of view of non-developers, yes, makes sense. An `if` statement is more readable than `deltemplates`